docs/learning/api-support-policies.md

```diff
@@ -69,7 +69,93 @@ Rare exceptions may be made to this policy when a breaking API change is require
 
 Most public APIs are intended to remain stable indefinitely, but occasionally a breaking change is required to fix a bug, introduce new functionality, or address performance problems. The deprecation policy ensures that such changes are introduced deliberately and predictably to help users plan ahead.
 
-Before a "public" API can be removed it must first be marked as "deprecated" in version `N` of the package. It must remain stable for the duration of version `N+1`. Finally, in version `N+2` it _may_ be removed. This means that, for example, an API deprecated in version 4.1 of a package cannot be removed until version 6.0 at the earliest.
+Before a "public" API can be removed it must first be marked as "deprecated", with a clear upgrade path described in the deprecation note and the earliest date when it might be removed. A grace period of one year is given for consumers to update their code. After the grace period, deprecated APIs may be removed in any following major release. We recommend using [typescript-eslint](https://typescript-eslint.io/) with the [no-deprecated rule](https://typescript-eslint.io/rules/no-deprecated/) enabled so that you get timely warnings when you use deprecated APIs.
+
+The same policy applies to "preview" APIs, with a 3 month grace period instead of a year.
```

Comment:
The way I understand it is that a Preview API could be added and removed within the same major version. Can someone confirm?

Also, I am assuming here we won't remove a Preview API in a patch?

Reply:
@aruniverse if beta APIs can be removed anytime (I was under the impression that they also had to wait for major release when writing this), do we need soft-deprecation (and deprecation comments) for preview APIs at all?